library/core/src/slice/mod.rs

```diff
@@ -2480,20 +2480,17 @@ impl<T> [T] {
             // coupled with the `left + size <= self.len()` invariant means
             // we have `left + size/2 < self.len()`, and this is in-bounds.
             let cmp = f(unsafe { self.get_unchecked(mid) });
-
-            // The reason why we use if/else control flow rather than match
-            // is because match reorders comparison operations, which is perf sensitive.
-            // This is x86 asm for u8: https://rust.godbolt.org/z/8Y8Pra.
-            if cmp == Less {
-                left = mid + 1;
-            } else if cmp == Greater {
-                right = mid;
-            } else {
-                // SAFETY: same as the `get_unchecked` above
-                unsafe { crate::intrinsics::assume(mid < self.len()) };
-                return Ok(mid);
+            
+            match cmp {
+                Less => left = mid + 1,
+                Greater => right = mid,
+                Equal => {
```

Comment:
Unfortunately this `match` is considered ambiguous, you likely need
```suggestion
                Ordering::Less => left = mid + 1,
                Ordering::Greater => right = mid,
                Ordering::Equal => {
```

Reply:
Thanks for your suggestion.

Reply:
Thankfully, the lint for specifically this issue will soon be deny-by-default thanks to #104154.

But, interestingly, I think this actually would have worked without the type -- Less and Greater are imported

https://github.com/rust-lang/rust/blob/e08b379d5d5840a2976d73d1754e8821a9973b66/library/core/src/slice/mod.rs#L9

and Equal is third so it being a catch-all would work.

Definitely better to fix the warnings, though 🙃